src/models/tools/geometry/jxg-board.ts

```diff
@@ -152,7 +158,8 @@ export const boardChangeAgent: JXGChangeAgent = {
     // If we are undoing an action, then the board already exists but its axes have
     // been removed, so we have to add the axes in that case as well.
     const updatedBoundingBox = scaleBoundingBoxToElement(board.containerObj.id, props);
-    const axes = addAxes(board, board.unitX, board.unitY, updatedBoundingBox);
+    const [unitX, unitY] = getAxisUnits(props);
```

Comment:
This was the fix - previously, if you updated `unitX` and `unitY` and then performed an undo, `unitX` and `unitY` were not updated, so the grid lines were drawn incorrectly.